src/analysis/retail/druid/balance/modules/spells/TouchTheCosmos.tsx

```diff
@@ -0,0 +1,155 @@
+  totcActivationTimestamp: number | null = null;
+  totcConsumptionTimestamp: number | null = null;
+  listCastStarsurge: number[];
+  listCastStarsurgeTimestamps: number[];
+  savedAP = 0;
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.has4PieceByTier(TIERS.T29);
+    this.listCastStarsurge = [];
+    this.listCastStarsurgeTimestamps = [];
+
+    Object.values(TOUCH_COSMOS.AFFECTED_CAST).forEach((spell) => {
+      this.totcBuffedAbilities[spell.id] = 0;
+    });
+
+    this.addEventListener(
+      Events.cast.by(SELECTED_PLAYER).spell(TOUCH_COSMOS.AFFECTED_CAST),
+      this.onCast,
+    );
+    this.addEventListener(
+      Events.damage.by(SELECTED_PLAYER).spell(TOUCH_COSMOS.AFFECTED_DAMAGE),
+      this.onDamage,
+    );
+  }
+
+  onCast(event: CastEvent) {
+    const hasTouchCosmos = this.selectedCombatant.hasBuff(SPELLS.TOUCH_THE_COSMOS.id);
+    if (!hasTouchCosmos) {
+      return;
+    }
+
+    this.totcBuffedAbilities[event.ability.guid] += 1;
+    if (event.ability.guid === SPELLS.STARSURGE_MOONKIN.id) {
+      //Append a Starsurge CastEvent and keep its timestamp
+      this.listCastStarsurge.push(1);
+      this.listCastStarsurgeTimestamps.push(event.timestamp);
+      this.savedAP += TOUCH_COSMOS.STARSURGE_COST;
+    }
+    if (event.ability.guid === SPELLS.STARFALL_CAST.id) {
+      this.savedAP += TOUCH_COSMOS.STARFALL_COST;
+    }
+  }
+
+  onDamage(event: DamageEvent) {
+    let checkCastList = true;
+
+    //Starsurge
+    if (event.ability.guid === SPELLS.STARSURGE_MOONKIN.id) {
+      while (checkCastList) {
+        if (!this.listCastStarsurge) {
+          // No CastEvent remaining, discard this source of damage. Ex: Wraths from Convoke
+          return;
+        }
+        if (
+          event.timestamp < (this.listCastStarsurgeTimestamps[0] + TOUCH_COSMOS.STARSURGE_MIN || 0)
+        ) {
+          // DamageEvent before a CastEvent, discard this source of damage.
+          return;
+        }
+        if (
+          event.timestamp >
+          (this.listCastStarsurgeTimestamps[0] || Infinity) + TOUCH_COSMOS.STARSURGE_DELAY
+        ) {
+          // DamageEvent to long after a CastEvent, discard this CastEvent
+          // (this is why their is a while loop) and try the next one.
+          this.listCastStarsurge.shift();
+          this.listCastStarsurgeTimestamps.shift();
+        } else {
+          // DamageEvent and CastEvent are within an acceptable timeframe, match theem together
+          this.listCastStarsurge.shift();
+          this.totalDamage += calculateEffectiveDamage(event, TOUCH_COSMOS.DAMAGE_BONUS);
+          this.listCastStarsurgeTimestamps.shift();
+          checkCastList = false;
+        }
+      }
+    }
+    //Starfall
+    if (event.ability.guid === SPELLS.STARFALL.id) {
+      return;
+    }
+  }
+
+  statistic() {
+    const dpsAdded = this.totalDamage / (this.owner.fightDuration / 1000);
+    return (
+      <Statistic
```

Comment:
one more nit I realized looking at the 2P code, cant you add it to the items category?